presto-main/src/test/java/com/facebook/presto/type/TestBlockPosition.java

```diff
@@ -0,0 +1,91 @@
+package com.facebook.presto.type;
+
+import com.facebook.presto.metadata.BoundVariables;
+import com.facebook.presto.metadata.FunctionKind;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.operator.scalar.AbstractTestFunctions;
+import com.facebook.presto.operator.scalar.ScalarFunctionImplementation;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.ArgumentProperty.valueTypeArgumentProperty;
+import static com.facebook.presto.operator.scalar.ScalarFunctionImplementation.NullConvention.BLOCK_POSITION;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.type.TestBlockPosition.BlockPosition.BLOCK_POSITION_IMPLEMENTATION;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class TestBlockPosition
+    extends AbstractTestFunctions
+{
+    @BeforeClass
+    public void setUp()
+    {
+        registerScalarFunction(BLOCK_POSITION_IMPLEMENTATION);
+    }
+
+    @Test
+    public void testBlockPosition()
+    {
+        assertFunction("identity(1)", BIGINT, 1);
```

Comment:
I understand that this test would fails today. I will revisit when you make the additional necessary changes.